geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/ListAsyncEventQueuesCommand.java

```diff
@@ -15,104 +15,105 @@
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.springframework.shell.core.annotation.CliCommand;
 
-import org.apache.geode.SystemFailure;
-import org.apache.geode.cache.execute.ResultCollector;
 import org.apache.geode.distributed.DistributedMember;
-import org.apache.geode.management.cli.Result;
-import org.apache.geode.management.internal.cli.CliUtil;
 import org.apache.geode.management.internal.cli.domain.AsyncEventQueueDetails;
 import org.apache.geode.management.internal.cli.functions.CliFunctionResult;
 import org.apache.geode.management.internal.cli.functions.ListAsyncEventQueuesFunction;
 import org.apache.geode.management.internal.cli.i18n.CliStrings;
-import org.apache.geode.management.internal.cli.result.ResultBuilder;
-import org.apache.geode.management.internal.cli.result.TabularResultData;
+import org.apache.geode.management.internal.cli.result.model.ResultModel;
+import org.apache.geode.management.internal.cli.result.model.TabularResultModel;
 import org.apache.geode.management.internal.security.ResourceOperation;
 import org.apache.geode.security.ResourcePermission;
 
 public class ListAsyncEventQueuesCommand extends InternalGfshCommand {
+  private static final String[] DETAILS_OUTPUT_COLUMNS =
+      {"Member", "ID", "Batch Size", "Persistent", "Disk Store", "Max Memory", "Listener"};
+  private static final String ASYNC_EVENT_QUEUES_TABLE_SECTION = "Async Event Queues";
+  private static final String MEMBER_ERRORS_TABLE_SECTION = "Member Errors";
+
   @CliCommand(value = CliStrings.LIST_ASYNC_EVENT_QUEUES,
       help = CliStrings.LIST_ASYNC_EVENT_QUEUES__HELP)
   @ResourceOperation(resource = ResourcePermission.Resource.CLUSTER,
       operation = ResourcePermission.Operation.READ)
-  public Result listAsyncEventQueues() {
-    try {
-      TabularResultData tabularData = ResultBuilder.createTabularResultData();
-      boolean accumulatedData = false;
+  public ResultModel listAsyncEventQueues() {
+    Set<DistributedMember> targetMembers = getAllNormalMembers();
+    if (targetMembers.isEmpty()) {
+      return ResultModel.createError(CliStrings.NO_MEMBERS_FOUND_MESSAGE);
+    }
 
-      Set<DistributedMember> targetMembers = getAllNormalMembers();
+    // Each (successful) member returns a list of AsyncEventQueueDetails.
+    List<CliFunctionResult> results = executeAndGetFunctionResult(
+        new ListAsyncEventQueuesFunction(), new Object[] {}, targetMembers);
 
-      if (targetMembers.isEmpty()) {
-        return ResultBuilder.createUserErrorResult(CliStrings.NO_MEMBERS_FOUND_MESSAGE);
-      }
+    ResultModel result = buildAsyncEventQueueInfo(results);
 
-      ResultCollector<?, ?> rc = CliUtil.executeFunction(new ListAsyncEventQueuesFunction(),
-          new Object[] {}, targetMembers);
-      List<CliFunctionResult> results = CliFunctionResult.cleanResults((List<?>) rc.getResult());
+    // Report any explicit errors as well.
+    if (results.stream().anyMatch(r -> !r.isSuccessful())) {
```

Comment:
Any reason we want to list "success" and "error" in two different tables? All other commands are putting them in one table.